lisa/operating_system.py

```diff
@@ -282,15 +282,20 @@ def install_packages(
         package_names = self._get_package_list(packages)
         self._install_packages(package_names, signed)
 
-    def package_exists(
-        self, package: Union[str, Tool, Type[Tool]], signed: bool = True
-    ) -> bool:
+    def package_installed(self, package: Union[str, Tool, Type[Tool]]) -> bool:
         """
         Query if a package/tool is installed on the node.
         Return Value - bool
         """
         package_name = self.__resolve_package_name(package)
-        return self._package_exists(package_name)
+        return self._package_installed(package_name)
+
+    def package_existed(self, package: str) -> bool:
+        """
+        Query if a package/tool exists in the repo
+        Return Value - bool
+        """
+        return self._package_existed(package)
```

Comment:
Does it need to call `package_name = self.__resolve_package_name(package)`?

Reply:
to be frank, I am not clear what is the usage of __resolve_package_name

Reply:
Some packages are passed in as Tool type, not a string. The method will convert it to a package str by calling the `package_name` method.

Reply:
updated